library/std/src/sys/unix/fd.rs

```diff
@@ -256,3 +272,16 @@ impl Drop for FileDesc {
         let _ = unsafe { libc::close(self.fd) };
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{FileDesc, IoSlice};
+
+    #[test]
+    fn limit_vector_count() {
+        let stdout = FileDesc { fd: 0 };
```

Comment:
Standard output is fd 1. 0 is standard input.

Reply:
Indeed. Interesting that writing to stdin does not fail. (I did test this before and after to check that it triggers the issue here and is fixed by the fix.) Will fix.

Reply:
Fixed. I also checked and
```
use std::fs::File;
use std::io::Write;
use std::os::unix::io::FromRawFd;

fn main() {
    unsafe { File::from_raw_fd(0) }.write_all(b"FOO\n").unwrap();
}
```
write happily write to stdin which appears in the terminal just like stdout but indeed is not captured when e.g. redirecting stdout. Today I learned...